### PR TITLE
Implement timing metrics

### PR DESCRIPTION
**Pull request by vkitchen**
Measures the time taken to perform the search inside the container
Measures the time taken to load the index in a new container
Reports the two sets of timing information, and the difference between them
A solution to #23

### CODE REVIEW

File: searcher.py [hunk 1]
```diff
@@ -45,13 +47,69 @@ def search(self, client, output_path_guest, topic_path_host, topic_path_guest, g
...
+        result_minutes = []
+        result_seconds = []
+        for i in range(len(load_times)):
+            result_minutes.append(int(search_times[i].group(2)) - int(load_times[i].group(2)))
+            result_seconds.append(float(search_times[i].group(3)) - float(load_times[i].group(3)))
+        print('**********')
+        print('Search timing less load')
+        print('real\t{}m{:.3f}s'.format(result_minutes[0], result_seconds[0]))
+        print('user\t{}m{:.3f}s'.format(result_minutes[1], result_seconds[1]))
+        print('sys\t{}m{:.3f}s'.format(result_minutes[2], result_seconds[2]))
```
Comment (r-clancy) [inline searcher.py]: Small nit can a space be added so the formatting is consistent between the output of `time`? Might be useful for parsing the output later if needed.

```
**********
Search timing information
real    0m 13.72s
user    0m 17.58s
sys     0m 0.43s

**********
Search timing less load
real    0m12.680s
user    0m15.300s
sys     0m0.330s
```

File: searcher.py [hunk 2]
```diff
@@ -45,13 +47,69 @@ def search(self, client, output_path_guest, topic_path_host, topic_path_guest, g
...
+            single_query = queries[:query_end+6]
+            single_query_file = os.path.splitext(os.path.basename(self.config.topic))[0] + '.single.txt'
+            out_file = open(os.path.join(topic_path_host, single_query_file), 'w')
+            out_file.write(single_query)
+            out_file.close()
+
+        # Time empty search
+        search_args['topic']['path'] = os.path.join(topic_path_guest, single_query_file)
+        container = client.containers.run("{}:{}".format(self.config.repo, save_tag),
+                                          command="bash -c 'time /search --json {}'".format(json.dumps(json.dumps(search_args))), volumes=volumes, detach=True)
```
[the remark this answers is not in the document]
Comment (vkitchen) [reply searcher.py]: Some of the images which are based on `Debian` or derivative distributions have `bash` (which contains a builtin `time`) installed but no system wide `time` command. The `sh` also tends to be linked to `dash` in these images (which doesn't contain the builtin). I'm working on a solution now which will use either `time` or the builtin in `bash` depending on what is available. I'll update the pull request when that's done

[the remark this answers is not in the document]
Comment (r-clancy) [reply searcher.py]: Ahh, gotcha - once you've added that we'll merge this PR.

**Comment by r-clancy** [comment]
@vkitchen Can we make this optional by adding a `--timings` flag to the `search` hook? We can add in the `README` that we expect `bash` (or `time`) to be present in the image. This way, we don't need to do any checks for what's available in the image and only people who are interested in the timings need to worry about having support in their image.

**Comment by lintool** [comment]
+1 on making timings optional!

**Pull request merged**